spring-cloud-alicloud-acm/src/main/java/com/alibaba/alicloud/acm/AcmPropertySourceRepository.java

```diff
@@ -17,54 +17,43 @@
 package com.alibaba.alicloud.acm;
 
 import java.util.ArrayList;
+import java.util.Collection;
 import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArraySet;
 
 import com.alibaba.alicloud.acm.bootstrap.AcmPropertySource;
 
-import org.springframework.context.ApplicationContext;
-import org.springframework.context.ConfigurableApplicationContext;
-import org.springframework.core.env.CompositePropertySource;
 import org.springframework.core.env.PropertySource;
 
 /**
  * @author juven.xuxb, 5/17/16.
  */
 public class AcmPropertySourceRepository {
 
-	private final ApplicationContext applicationContext;
+	private static final Set<AcmPropertySource> ACM_PROPERTY_SOURCE_REPOSITORY = new CopyOnWriteArraySet();
+
+	private AcmPropertySourceRepository () {
```

Comment:
The `PropertySourceRepository` should not be used like a tool class. Violated the original intention of `Repository`.

Reply:
yes, the name is really not very reasonable, as `NacosPropertySourceRepository` do.

Reply:
`DubboServiceMetadataRepository` is a better example.